app/setup.py

```diff
@@ -324,6 +327,22 @@ def setup_submitter(application):
         raise Exception("Unknown EQ_SUBMISSION_BACKEND")
 
 
+def setup_publisher(application):
+    if application.config["EQ_PUBLISHER_BACKEND"] == "pubsub":
+        topic_id = application.config.get("EQ_PUB_SUB_TOPIC_ID")
```

Comment:
The topic id should be used on publishing rather than on initialisation. The publisher could be used to publish to multiple topics.

Reply:
Have added as part of the ABC changes. Also, I have kept the logic of raise an error if the fulfilment topic env var is not set. However, I could also handle it not being set on publish if we didn't want to do it on start up.